openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -16,11 +16,31 @@
 from openmdao.util.record_util import create_local_meta, update_local_meta
 
 
+def _check_imports():
+    """ Dynamically remove optimizers we don't have
+    """
+
+    optlist = ['ALPSO', 'CONMIN', 'FSQP', 'IPOPT', 'NLPQLP',
+               'NSGA2', 'PSQP', 'SLSQP', 'SNOPT', 'NLPY_AUGLAG', 'NOMAD']
+
```

Comment:
I've only tested SNOPT and IPOPT. There were a few minor fixes to the IPOPT wrapper that I made a pull request on to pyoptsparse. I've not yet tested the rest.